girder/stemserver_plugin/models/stemimage.py

```diff
@@ -130,3 +150,35 @@ def bright_shape(self, id, user):
 
     def dark_shape(self, id, user):
         return self._get_h5_dataset_shape(id, user, '/stem/dark')
+
+    def _get_import_stem_item(self, user):
+        """Get the item where we will store stem images
+
+        A new folder and item will be created if they do not exist.
+        Otherwise, they will be reused.
+
+        """
+        folder = FolderModel().createFolder(user, StemImage.IMPORT_FOLDER,
+                                            parentType='user', public=False,
+                                            creator=user, reuseExisting=True)
+
+        return ItemModel().createItem(StemImage.IMPORT_ITEM_NAME, user, folder,
+                                      reuseExisting=True)
+
+    def _get_import_unique_file_name(self, item):
+        """Get a unique name of a file to go in the item"""
+        files = ItemModel().childFiles(item)
+        names = [x['name'] for x in files]
+        for i in range(1, 1000):
+            if str(i) not in names:
+                return str(i)
+
+        raise RestException('Too many files in item:\n' + str(item), code=400)
```

Comment:
Lets not restrict the number of files and just using the filename for the filename ( it doesn't have to be unique )!

Reply:
Well, actually, it looks like they set [reuseExisting to be true](https://github.com/girder/girder/blob/292690e7e4c269ed3b34757ba86ddfa2713f9f16/girder/utility/filesystem_assetstore_adapter.py#L341) for the imported file. This makes if you try to import a file that has the same name as a file already existing, it just reuses the existing file.

We might have to make a new item for every file to use the file name. I'll try it out.

Reply:
Ok, thanks

Reply:
Yeah, apparently we can just make a new item for each one. It just puts a '(1)' or similar at the end if the name is the same.

![Screenshot from 2019-04-30 14-55-44](https://user-images.githubusercontent.com/9558430/56986177-14b1c700-6b58-11e9-9b35-d64436dd8745.png)

Reply:
Okay. I fixed it so that we make a new item for every imported file now, and we now use the filename for the item name.